packages/fast-tooling-react/src/css-property-editor/property-editor.tsx

```diff
@@ -0,0 +1,270 @@
+import {
+    CSSProperties,
+    CSSPropertiesConfig,
+    CSSPropertyConfig,
+    CSSPropertyEditorHandledProps,
+    CSSPropertyEditorProps,
+    CSSPropertyEditorState,
+    CSSPropertyEditorUnhandledProps,
+} from "./property-editor.props";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+
+export default class CSSPropertyEditor extends Foundation<
+    CSSPropertyEditorHandledProps,
+    CSSPropertyEditorUnhandledProps,
+    CSSPropertyEditorState
+> {
+    public static displayName: string = "CSSPropertyEditor";
+
+    protected handledProps: HandledProps<CSSPropertyEditorHandledProps> = {
+        data: void 0,
+        onChange: void 0,
+        managedClasses: void 0,
+    };
+
+    private propertyEditorRef: React.RefObject<HTMLPreElement>;
+
+    private keyInputRef: React.RefObject<HTMLInputElement>;
+
+    constructor(props: CSSPropertyEditorProps) {
+        super(props);
+
+        this.keyInputRef = React.createRef();
+        this.propertyEditorRef = React.createRef();
+
+        this.state = {
+            key: "",
+            value: "",
+            data: this.getCSSPropertyConfig(this.props.data),
+        };
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <pre
+                className={this.props.managedClasses.cssPropertyEditor}
+                onClick={this.handleClick}
+                ref={this.propertyEditorRef}
+            >
+                <div
+                    className={this.props.managedClasses.cssPropertyEditor_propertyRegion}
+                >
+                    {this.renderCurrentProperties()}
+                    {this.renderNewProperty()}
+                </div>
+            </pre>
+        );
+    }
+
+    public componentDidUpdate(prevProps: CSSPropertyEditorProps): void {
+        if (prevProps.data !== this.props.data) {
+            this.setState({ data: this.getCSSPropertyConfig(this.props.data) });
+        }
+    }
+
+    private renderCurrentProperties(): React.ReactNode {
+        if (!this.state.data) {
+            return;
+        }
+
+        return Object.keys(this.state.data).map(
+            (cssKey: string, index: number): React.ReactNode => {
+                return this.renderPropertyInput(
+                    cssKey,
+                    this.state.data[cssKey].value,
+                    this.state.data[cssKey].keyWidth,
+                    this.state.data[cssKey].valueWidth,
+                    index
+                );
+            }
+        );
+    }
+
+    private renderNewProperty(): React.ReactNode {
+        const newPropertyConfig: CSSPropertiesConfig = this.getCSSPropertyConfig({
+            [this.state.key]: this.state.value,
+        });
+
+        return this.renderNewPropertyInput(
+            newPropertyConfig[this.state.key].keyWidth,
+            newPropertyConfig[this.state.key].valueWidth
+        );
+    }
+
+    private renderPropertyInput(
+        key: string,
```

Comment:
This function would be more simple if it accepted the config object and not all the data from the config object as individual arguments